types/src/transaction/mod.rs

```diff
@@ -297,13 +309,24 @@ impl RawTransaction {
 
 #[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
 pub enum TransactionPayload {
-    WriteSet(ChangeSet),
+    WriteSet(WriteSetPayload),
     /// A transaction that executes code.
     Script(Script),
     /// A transaction that publishes code.
     Module(Module),
 }
 
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum WriteSetPayload {
+    /// Directly passing in the write set.
+    Direct(ChangeSet),
+    /// Generate the writeset by running a script.
+    WriteSetScript {
```

Comment:
Nit: it's a bit asymmetric to have one called `WriteSetPayload::Direct` and one called `WriteSetPayload::WriteSetScript`. Maybe just `WriteSetPayload::Script`?